### PR TITLE
fix python3 compatibility

### PR DESCRIPTION
**Pull request by spaceone**

### CODE REVIEW

**Comment by prologic** [comment]
I was trying to fix this earlier.

This needs a ``from circuits.six import exec_``.

There is no ``exec()`` function in Python 2.6

**Comment by spaceone** [comment]
probably not possible to import from circuits at that point!

**Comment by prologic** [comment]
Okay; then this is going to have to have some rework I think as it breaks pretty much everything I think? :)

**Comment by spaceone** [comment]
Well, now it works :)

**Comment by prologic** [comment]
Awesome ☺

File: circuits/__init__.py [hunk 1]
```diff
@@ -36,6 +37,7 @@
 	for _path in __path__:
 		_path = os.path.join(_path, '__init__.py')
 		if _path != __file__ and os.path.exists(_path):
+			from .six import exec_
 			with open(_path) as fd:
-				exec fd in globals()
+				exec_(fd, globals())
 	del os, extend_path, _path, fd
```
Comment (prologic) [inline circuits/__init__.py]: Need a ``del exec_`` here too :)

**Comment by spaceone** [comment]
del exec_ done

**Comment by prologic** [comment]
Merged despite failing tests. I think they're failing for other reasons though. The new code changes that allow you to disable the circuits.web advertising seems to be breaking a few things :)

**Pull request merged**